dask/bag/core.py

```diff
@@ -380,52 +380,126 @@ def __str__(self):
 
     str = property(fget=StringAccessor)
 
-    def map(self, func, **kwargs):
-        """ Map a function across all elements in collection
+    def map(self, func, *args, **kwargs):
+        """Apply a function elementwise across one or more bags.
 
+        Note that all ``Bag`` arguments must have the same number of partitions.
```

Comment:
Perhaps "must be partitioned identically".  This operation will not realign partitions.